src/views/v3/Bridge/Routes/index.tsx

```diff
@@ -51,16 +54,93 @@ function Routes({
   const [originalSelectedRoute, setOriginalSelectedRoute] = useState<
     string | undefined
   >(selectedRoute);
+  const [selectedRouteBadge, setSelectedRouteBadge] =
+    useState<string>('fastest');
+
+  const routesWithQuotes = routesList.filter((rs) => quotes[rs] !== undefined);
+
+  const selectedQuote = !selectedRoute
+    ? undefined
+    : (() => {
+        const quoteResult = quotes[selectedRoute];
+        return quoteResult?.success ? quoteResult : undefined;
+      })();
+
+  const fastestRoute = routesWithQuotes.reduce(
+    (fastest, route) => {
+      const quote = quotes[route];
+      if (!quote || !quote.success) return fastest;
+
+      if (quote.eta !== undefined && quote.eta < fastest.eta) {
+        return { name: route, eta: quote.eta };
+      }
+      return fastest;
+    },
+    { name: '', eta: Infinity },
+  );
+
+  const cheapestRoute = routesWithQuotes.reduce(
```

Comment:
I think we should move these functions out of the render/component logic as they aren't directly related to react rendering. And I think you can reduce 3 loops to 1. Line 60, 69, 82 can be combined.

```ts
function getFastestAndCheapest(routesList) {
  const cheapest = ...
  const fastest = ...

  for (const route of routesList) {
    ...
  }
}
```

Optionally, you can [fast-memoize](https://www.npmjs.com/package/fast-memoize) this function if there is a need if routeList doesn't change too often but parent component renders more often.